files/build_templates/docker_image_ctl.j2

```diff
@@ -144,6 +144,7 @@ start() {
     # TODO: Mellanox will remove the --tmpfs exception after SDK socket path changed in new SDK version
 {%- endif %}
     docker create {{docker_image_run_opt}} \
+        --uts=host \{# W/A: this should be set per-docker, for those dockers which really need host's UTS namespace #}
```

Comment:
Can you think of an instance where we would _not_ want to share the host's UTS namespace?

Reply:
I cannot :smile: 
This is a modified cherry-pick following master branch.

---
In reply to: [387377484](https://github.com/Azure/sonic-buildimage/pull/4219#discussion_r387377484) [](ancestors = 387377484)

Reply:
:smile: Then maybe add a "maybe" to the beginning of the comment?

Reply:
@jleveque @qiluo-msft I think swss, syncd, pmon don't need to care about the hostname, if they do IMO it looks like a design mistake. The idea behind docker containers is isolation, why do we need docker containers at all if we want to share and map inside container almost everything?
Besides, --priviledge + --uts=host will give container priviledges to change host OS hostname.
The only containers I know which need host's hostname inside container are snmp and dhcp_relay but it doesn't look like they need --priviledge option or do they?
I would like to keep this comment and revise containers isolation in sonic in general in the future.